### PR TITLE
style(suite): Redesign banners

### LINKED ISSUE #10523: Review: Restyle yellow "Learn more" button
- It should follow theme colors: probably `backgroundAlertYellowBold`
- we should use our `Button` component with specific `size`

Screenshot 2024-01-05 at 14 32 46

It shouldn't be underlined on hover:
Screenshot 2024-01-08 at 14 45 18

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve #10523

### new Button variants

**info**
image

image

**warning**
image

image

### NotificationCard

Note: Button backgrounds are not perfect becase we don't have defined all variantions for button backgrounds on different elevations. It will be done in different PR.

**info**

Screenshot 2024-01-24 at 14 36 40

Screenshot 2024-01-24 at 14 36 46

image

image

**warning**
Screenshot 2024-01-24 at 14 33 56

Screenshot 2024-01-24 at 14 37 40

Screenshot 2024-01-24 at 15 58 08

image

**destructive**
Screenshot 2024-01-24 at 14 36 19

Screenshot 2024-01-24 at 14 37 03

Screenshot 2024-01-24 at 15 59 17

Screenshot 2024-01-24 at 15 59 22

### Banner

**info**
Screenshot 2024-01-24 at 15 17 07

Screenshot 2024-01-24 at 15 14 11
Screenshot 2024-01-24 at 15 59 17

**warning**

Screenshot 2024-01-24 at 15 24 57

Screenshot 2024-01-24 at 15 24 28

Screenshot 2024-01-24 at 15 31 34

**destructive**
Screenshot 2024-01-24 at 15 16 23

Screenshot 2024-01-24 at 15 14 56

### CODE REVIEW

**Review by peter-sanderson — APPROVED** [review]

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/NotificationCard.tsx]: This styled Button should not be there 🙏  My best guess is that the banner should render the [Warning](https://www.figma.com/file/Y2E8dcdNYrSMGIVRjYFXKp/%5Bs2.0%5D-Components?type=design&node-id=3172-19021&mode=dev) variant of the button, which hadn't been there at the time of me making the `Button` component, so it should be added first.

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/NotificationCard.tsx]: all those functions should also be replaced by simply using the right Button variants.

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/NotificationCard.tsx]: change the `Wrapper` BG to `backgroundTertiaryDefaultOnElevation0`

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/NotificationCard.tsx:25]: How about using the `ButtonVariant` as the `variant` type?

File: packages/suite/src/components/suite/banners/Banner.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/Banner.tsx:27]: I feel like that those BGs being bold might be out of place considering the current stylistic direction. Waiting for an answer from Ondra!

image

File: packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx:23]: I would rather dynamically add a border above the page, to accommodate the coinjoin status bars too :) Although that also does not look ideal but then the Banner change is out of scope anyways 😓 

image

**Review by dahaca — CHANGES_REQUESTED** [review]
Let's wait for an answer from Ondra or David and make the banners a proper part of the design system ;)

**Comment by dahaca** [comment]
### Update: 

* This might be the way to go for the critical banners, which matches the mobile version's approach, however it creates problems for the coinjoin status bars. They could be moved to the `Sidebar` but that's in the future. 

image

* And those are the mobile banners, likely to be the solution here.



*** 

Still no answer from Ondra but I can imagine that to not bloat the scope we won't make any drastic changes. Let's wait for Monday!

**Comment by jvaclavik** [comment]
> ### Update:
> * This might be the way to go for the critical banners, which matches the mobile version's approach, however it creates problems for the coinjoin status bars. They could be moved to the `Sidebar` but that's in the future.
> 
> image
> 
> * And those are the mobile banners, likely to be the solution here.
> 
> 
> Still no answer from Ondra but I can imagine that to not bloat the scope we won't make any drastic changes. Let's wait for Monday!

Do you have more info here?

I'm totally for rewriting banners from scratch because it's a mess and it would be very nice to have it in the design system. If we are going to rewrite it properly we might close this PR without merging, but at the same time, I think we don't have enough time for it before release and we will need to merge just some hacky solution like this.

**Comment by dahaca** [comment]
> Do you have more info here?

Ondra and David still did not come up with the final decision. I can ask again but for now I believe the most efficient thing for us is to fix the color bugs from the original issue and not do anything else until it's all properly designed and integrated in the system. Do you agree?

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/NotificationCard.tsx:25]: I don't think it's a good idea.

```
export type ButtonVariant =
    | 'primary'
    | 'secondary'
    | 'tertiary'
    | 'info'
    | 'warning'
    | 'destructive';
```

We have only: 
```
    'info'
    | 'warning'
    | 'destructive'
```

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (jvaclavik) [inline packages/suite/src/components/suite/NotificationCard.tsx:102]: `TrezorLink` blocks component from moving to `@trezor/components`. Any ideas here?

File: packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx:23]: It needs to be conditional if it's above the page, and I'm not sure if we can assure it.

Comment (jvaclavik) [reply packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx:23]: I think it's minor and we can create separate issue for it, WDYT?

File: packages/suite/src/components/suite/banners/Banner.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/banners/Banner.tsx:27]: It follows previous design. Anyway, I think there should be one component, called `Banner` (or whatever) and it should solve both variants (currently `NotificationCard` and `Banner`). Something like in Google material design: https://mui.com/material-ui/react-alert/

`variant: standard`:
image

`variant: filled`:
image

`variant: outlined`:
image


`MUI API`
image


WDYT?

Comment (dahaca) [reply packages/suite/src/components/suite/banners/Banner.tsx:27]: Totally! We have even more components fulfilling the role of a "banner", there are some in settings as you probably have seen! 

It definitely is one of the obvious candidates for an issue for the next phase of Empower and it's on the designers to set a clear goal.

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:25]: Then how about `Pick`ing?? 

Not in this PR but I believe it would be nice to create a`type UIVariant = 'primary'
    | 'secondary'
    | 'tertiary'
    | 'info'
    | 'warning'
    | 'destructive'` and `Pick` from it in all places where we have those variants ;)

File: packages/suite/src/components/suite/section/SectionItem.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/section/SectionItem.tsx]: This was added here for the anchor highlighting. The comment is not the clearest though, I agree. You did it because of the `Select` bug? Some additional styling is needed somewhere to get rid of this and not break the anchor highlighting. Better be handled it in a separate [issue](https://github.com/trezor/trezor-suite/issues/10792)!

File: packages/components/src/components/buttons/buttonStyleUtils.ts
Comment (peter-sanderson) [inline packages/components/src/components/buttons/buttonStyleUtils.ts:60]: optional: would be great to turn switch to simple Record map

File: packages/suite/src/components/suite/banners/Banner.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/Banner.tsx]: I believe blue should be the default? Also does not work in the light mode like that

image

File: packages/suite/src/components/suite/banners/Banner.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/Banner.tsx]: could you leave a TODO in the colors.ts? 👀

File: packages/suite/src/components/suite/banners/Banner.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/Banner.tsx]: nit: I'd use text inverse here _(and fix it in the colors)_

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:102]: The first thing that comes to mind is that it can get the Modal treatment, a `Banner` in suite with the `useExternalLink` in it, which received the correct URL and passes it to the `Banner` from components 🤷

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/NotificationCard.tsx:120]: can simply be `variant || 'tertiary'`?

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/NotificationCard.tsx:25]: It's `Extract` instead of `Pick`, but OK. Agree with all of it :)

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/NotificationCard.tsx:102]: OK, next PR

File: packages/components/src/components/buttons/buttonStyleUtils.ts
Comment (jvaclavik) [reply packages/components/src/components/buttons/buttonStyleUtils.ts:60]: Let's do it in another PR after merge/release

File: packages/suite/src/components/suite/section/SectionItem.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/section/SectionItem.tsx]: Makes sense, I deleted this change from this PR.

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (jvaclavik) [reply packages/suite/src/components/suite/NotificationCard.tsx:120]: I would rather keep it this way. `||` could be tricky. Let's improve it in PR about refactoring mapping values from tokens

Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:120]: No stress, we will go through the banners again once we make them a part of the design system!

Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:120]: And not sure what you mean by tricky, its type is `NotificationCardVariant | undefined`, so it should work fine, am I missing something?

File: packages/theme/src/colors.ts
Comment (dahaca) [inline packages/theme/src/colors.ts:146]: 👍

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:25]: https://github.com/trezor/trezor-suite/issues/10818

File: packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx
Comment (dahaca) [inline packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx:1]: One last nitpick, let's assign this section to `elevationNegative` for now, looks a bit cleaner IMO, no?

image

**Review by dahaca — APPROVED** [review]
At last 🎨

File: packages/suite/src/components/suite/NotificationCard.tsx
Comment (dahaca) [reply packages/suite/src/components/suite/NotificationCard.tsx:102]: Totally not now, we are not moving yet ;)

File: packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx
Comment (dahaca) [reply packages/suite/src/components/suite/banners/SuiteBanners/SuiteBanners.tsx:1]: And the `CoinjoinStatusBar`'s as well, I think it looks 🆗 

```
background: ${({ theme }) => theme.backgroundSurfaceElevationNegative};
border-bottom: 1px solid ${({ theme }) => theme.borderOnElevation0};
```

image

**Pull request merged**
